subsys/bluetooth/host/keys.c

```diff
@@ -371,3 +414,26 @@ SETTINGS_STATIC_HANDLER_DEFINE(bt_keys, "bt/keys", NULL, keys_set, keys_commit,
 			       NULL);
 
 #endif /* CONFIG_BT_SETTINGS */
+
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+void bt_keys_notify_usage(u8_t id, const bt_addr_le_t *addr)
+{
+	if (!addr) {
+		return;
+	}
+	struct bt_keys *keys = bt_keys_find_addr(id, addr);
+	if (!keys) {
+		return;
+	}
+	if (last_keys_updated == keys) {
+		return;
+	}
+	keys->aging_counter = ++aging_counter_val;
+	last_keys_updated = keys;
+	BT_DBG("Aging counter for %s is set to %u", bt_addr_le_str(addr),
+	       keys->aging_counter);
+#if defined(CONFIG_BT_KEYS_SAVE_AGING_COUNTER_ON_PAIRING)
+	bt_keys_store(keys);
```

Comment:
Correct me if I'm wrong, but this would not overwrite the key, but create a duplicate one. I think a bt_key_clear is be needed first.

Reply:
This will update the existing entry. According to [https://docs.zephyrproject.org/latest/reference/storage/settings/settings.html](url) "Deleting an existing key-value pair in the settings mean to set its value to NULL." There is no sense in setting the value to NULL and then to some correct value again. This will only increase flash usage.

Reply:
@nvlsianpu Is this correct. I cannot see from the documentation what would happen if settings_save_one is called on an existing entry. This comment is very confusing to me:
```
/*
 * Append a single value to persisted config. Don't store duplicate value.
 */
int settings_save_one(const char *name, const void *value, size_t val_len)
```
Does it mean we shouldn't store duplicate, or that the function won't store duplicate?

Reply:
> This will update the existing entry. According to https://docs.zephyrproject.org/latest/reference/storage/settings/settings.html "Deleting an existing key-value pair in the settings mean to set its value to NULL." There is no sense in setting the value to NULL and then to some correct value again. This will only increase flash usage.

This depends on how the key is stored, if the key is stored as "/bt/keys/10:00:07:08", it needs deletion or it will  stay in flash. Deleting the key "bt/keys/10:00..." ensures it is not kept when the "garbage collection" is done.

Reply:
> @nvlsianpu Is this correct? I cannot see from the documentation what would happen if settings_save_one is called on an existing entry. This comment is very confusing to me:
> /*
>  * Append a single value to persisted config. Don't store duplicate value.
>  */
> int settings_save_one(const char *name, const void *value, size_t val_len)
> 
> Does it mean we shouldn't store duplicate, or that the function won't store duplicate?

It means that the function won't store duplicates.

Reply:
Ok, thanks.